src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/PersonalSkillsAndInterestsEntityMapper.kt

```diff
@@ -50,8 +50,8 @@ abstract class PersonalSkillsAndInterestsEntityMapper {
   @Mapping(target = "skills", expression = "java( updateSkills(entity, dto) )")
   @Mapping(target = "interests", expression = "java( updateInterests(entity, dto) )")
   abstract fun updateEntityFromDto(
-    @MappingTarget entity: PersonalSkillsAndInterestsEntity,
-    dto: UpdatePersonalSkillsAndInterestsDto,
+    @MappingTarget entity: PersonalSkillsAndInterestsEntity?,
```

Comment:
Why have all the args for these various `updateEntity....` methods changed to nullable? 🤔

Reply:
In some cases they're not mandatory - for example if a question is never asked, so the entity won't exist.

I think I mentioned on Slack that I need to go through the cardinality of each field in more detail and do more thorough testing of the permutations. I don't like merging code that I know is going to be wrong in places, but this mapping code is quite involved as it is and I'd rather get it deployed and then test the various permutations. For example, what happens if an induction has qualifications, but they are then omitted in a subsequent update request. Are the original qualifications removed? There's lots to test!

Reply:
💯 👍 
OK, makes sense and happy with this approach, though I don't think the DTO arg would ever be null would it? Doesn't matter here and now though, if you are stepping back and looking at the cardinality of stuff separately then I'm sure it'll all get mopped up then 😁 👍